Assignment5/pablocabargas-21A328/assignment_notebook.py

```diff
@@ -0,0 +1,294 @@
+# ck.load_files_into_cache()
+
+"""# About this assignment.
+This assignment is based on https://github.com/usc-isi-i2/kgtk-notebooks/tree/main/tutorial. If you have any questions or doubts, it is encouraged to look how the tutorial performs the different operations.
+
+Additional information can be found in https://kgtk.readthedocs.io/
+
+## Simple graph statistics
+
+Let's calculate first some statistics about the KG. Count the number of instances:
+"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# kgtk("""
+#     query -i all
+#       --match '(a)-[:P31]->(class)'
+#       --return 'count(a) as N'
+#         
+# """)
+# 
+# 
+#
+
+"""Now, count the number of distinct properties: 
+
+"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# kgtk("""
+#     query -i all
+#       --match '()-[a]->()'
+#       --return 'count(distinct a.label) as N'
+# """)
+#
+
+"""Now, let's count the frequency of those properties. That is, how many instances we can find with each property"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# kgtk("""
+#    query -i all
+#     --match '(x)-[r]->()' \
+#     --return 'r.label as label, count(x) as frequency'
+# """)
+#
+
+"""## Simple queries
+Some of these queries are simple and will run in the Wikidata endpoint. 
+Try both of them using SPARQL and Kypher
+"""
+
+# Which actors has Schwarzenegger worked with throughout his career? (Print also the movie)
+
+# (in SPARQL)
+"""
+SELECT ?movie ?actors
+WHERE {
+  wd:Q2685 wdt:P800 ?movie.
+  ?movie wdt:P161 ?actors.
+}
+"""
+
+# In Kypher:
+kgtk("""
+    query -i all
+    --match '(:Q2685)-[:P800]->(movie)-[:P161]->(z)'
+    --return 'movie as movie, z as actors'
+""")
+
+# How many awards does Schwarzenegger have?
+
+# SPARQL:
+"""
+SELECT (COUNT(?premios) as ?cantidad_premios)
+WHERE {
+  wd:Q2685 wdt:P166 ?premios.
+}
+"""
+
+# Kypher:
+kgtk("""
+    query -i all
+      --match '(:Q2685)-[:P166]->(premios)'
+      --return 'count(premios) as premios'
+""")
+
+# Retrieve at least two members of Schwarzenegger's political party. Make sure only persons are returned
+# SPARQL:
+# TO DO
+"""
+SELECT DISTINCT ?member ?label
+WHERE {
+      wd:Q2685 wdt:P102 ?party.
+      ?member wdt:P102 ?party.
+      ?member wdt:P31 wd:Q5.
+      ?member rdfs:label ?label.
+      FILTER(LANG(?label) = 'en').
+}  """
+
+
+# Kypher:
+kgtk("""
+    query -i all
+    --match '(:Q2685)-[:P102]->(party)-[:P488]->(persons_of_chair)-[:P31]->(:Q5)'
+    --return 'party as party, persons_of_chair as persons_of_chair'"""
+)
+
+# What are the properties that describe an artist?
+
+# In theory this one is heavy on Wikidata
+
+# SPARQL: 
+# TO DO
+"""
+SELECT DISTINCT ?property
+WHERE {
+      ?subclasses rdfs:subClassOf* wd:Q483501.
+      ?x wdt:P106 ?subclasses.
+      ?x ?property ?v.
+} 
+"""
+
+
+
+# In Kypher:
+kgtk("""
+    query -i all
+    --match '()<-[p {label: property}]-()-[:P279*]->(:Q483501)'
```

Comment:
Not correct. You are showing the properties of the class, not from the individuals with that profession (P106)